### PR TITLE
docs(operators): Add documentation for bufferToggle

### LINKED ISSUE #112: Add docs for operator : transformation - bufferToggle
Add docs for operator bufferToggle under group transformation 
 
Below is minimum details needed 
 
- [ ] name 
- [ ] operatorType 
- [ ] signature 
- [ ] parameters 
- [ ] short description 
- [ ] Technical description 
- [ ] Visual example  (Marble diagram) 
- [ ] Real world example (Non-framework) 
- [ ] Related Operators

### PR DESCRIPTION
**Pull request by DiedrikDM**
Add documentation for bufferToggle to close #112

### CODE REVIEW

File: src/operator-docs/transformation/bufferToggle.ts [hunk 1]
```diff
@@ -1,6 +1,94 @@
...
+      import { empty } from 'rxjs/observable/empty';
+      import { map, bufferToggle } from 'rxjs/operators';
+
+      const clicks$ = fromEvent(document, 'click');
+      const openings$ = interval(1000);
+      const buffered$ = clicks$.pipe(
+        map(e => {return {X: e.clientX, Y: e.clientY};}),
+        bufferToggle(openings$, i => i % 2 ? interval(500) : empty())
+      );
+      buffered$.subscribe(x => console.log(x));
```
Comment (sumitarora) [inline src/operator-docs/transformation/bufferToggle.ts]: Please add expected output for the examples also.

Comment (DiedrikDM) [reply src/operator-docs/transformation/bufferToggle.ts]: @sumitarora  I don't really understand what you mean... Should I add a comment that explains what you'll see when you run this code? Or should I take some output from when I run it and put that in comment together with this code?

Comment (sumitarora) [reply src/operator-docs/transformation/bufferToggle.ts]: @DiedrikDM See the output block as part of code explaining expected output. https://github.com/ReactiveX/rxjs-docs/blob/master/src/operator-docs/combination/combineLatest.ts#L62

Comment (DiedrikDM) [reply src/operator-docs/transformation/bufferToggle.ts]: @sumitarora ok, I'll do that :) 👍

File: package.json [hunk 2]
```diff
@@ -48,7 +48,7 @@
     "zone.js": "0.8.14"
   },
   "devDependencies": {
-    "@angular/cli": "1.6.0",
+    "@angular/cli": "1.6.5",
```
Comment (frederikprijck) [inline package.json]: Any reason why this is changed in multiple PR's ?

File: src/operator-docs/transformation/bufferToggle.ts [hunk 3]
```diff
@@ -1,6 +1,147 @@
...
+      const keyUp = fromEvent(document,'keyup');
+      const openings = keyUp.pipe(filter(e => e.key === 's'));
+      const closing = keyUp.pipe(filter(e => e.key === 'e'));
+      const buffered = clicks.pipe(
+        bufferToggle(openings, _ => closing)
+      );
+      /*
+      Expected console output:
+
+      [[object Object] {
```
Comment (btroncone) [inline src/operator-docs/transformation/bufferToggle.ts]: Do we want to do expected output above or below the subscription? I think we have gone back and forth in examples thus far, we probably should be consistent moving forward. @sumitarora @ashwin-sureshkumar @ladyleet

Comment (ashwin-sureshkumar) [reply src/operator-docs/transformation/bufferToggle.ts]: Expected output should be below the subscription.

Comment (DiedrikDM) [reply src/operator-docs/transformation/bufferToggle.ts]: @btroncone @ashwin-sureshkumar guys... really :( ? I looked at the example from @sumitarora  and thought that it should've been above the subscription...
I'll change it in my PRs! 👍

**Pull request merged**